tests/SDAccel/hls_with_cppkernel_opencl_kernel_ap_vector_add_args.cpp

```diff
@@ -0,0 +1,88 @@
+/* Simple OpenCL vector addition using SYCL in OpenCL interoperability
+   mode Boost.Compute C++ host API and precompiled kernel.
+
+   Compared to using plain Boost.Compute OpenCL host API, in SYCL the
+   buffer transfers between host and device are implicits with the
+   accessors.
+*/
+
+/* REQUIRES: xilinx-xocc
+   RUN: %{execute}%s | %{filecheck} %s
+   CHECK: Result:
+   CHECK-NEXT: 6 8 11
+*/
+#include <iostream>
+#include <iterator>
+#include <boost/compute.hpp>
+#include <boost/test/minimal.hpp>
+
+#include <CL/sycl.hpp>
+
+//#include "cnl/FPGA/Xilinx/include/ap_int.h"
+//#include "/proj/xbuilds/2017.2_sdx_qualified_latest/installs/lin64/SDx/2017.2/Vivado_HLS/include/ap_int.h"
+#include "ap_int.h"
+
+using namespace cl::sycl;
+
+constexpr size_t N = 3;
+using Vector = ap_int<256>[N];
+
+int test_main(int argc, char *argv[]) {
+  Vector a = { 1, 2, 3 };
+  Vector b = { 5, 6, 8 };
+  Vector c;
+
+  /* Construct the queue from the default OpenCL one.
+
+     You can use the following environment variables to select
+     the device to be chosen at runtime
+     BOOST_COMPUTE_DEFAULT_DEVICE
+     BOOST_COMPUTE_DEFAULT_DEVICE_TYPE
+     BOOST_COMPUTE_DEFAULT_PLATFORM
+     BOOST_COMPUTE_DEFAULT_VENDOR
+
+     for example doing in bash
+     export BOOST_COMPUTE_DEFAULT_VENDOR=Xilinx
+     will probably select for execution a Xilinx FPGA on the machine
+  */
+  auto opencl_q { boost::compute::system::default_queue() };
+  queue q { opencl_q };
+
+  // Construct an OpenCL program from the precompiled kernel file
+  auto program =
+    boost::compute::program::create_with_binary_file("hls_ap_vector_add_kernel.xclbin",
+                                                     opencl_q.get_context());
+
+  // Build the OpenCL program
+  program.build();
+
+  // Build a SYCL kernel from the OpenCL kernel
+  kernel k { boost::compute::kernel { program, "vector_add" } };
+
+
+  // Create buffers from a & b vectors
+  buffer<ap_int<256>> A { std::begin(a), std::end(a) };
```

Comment:
In C++17 we will write just
```
buffer A { std::begin(a), std::end(a) };
```
but in the meantime you can write everywhere
```
buffer<Vector::value_type> A { std::begin(a), std::end(a) };
```
to have generic code just parameterized by the Vector type.

Reply:
Was it possible to use `buffer<Vector::value_type>` instead of repeating `ap_int<256>` everywhere ?